js/views/chatview.js

```diff
@@ -242,6 +240,29 @@
 			}
 		},
 
+		_getDateSeparator: function(timestamp) {
+			var date = moment(timestamp, 'x'),
+				today = moment(),
+				dayOfYear = OC.Util.formatDate(date, 'YYYY-DDD'),
+				dateOfToday = OC.Util.formatDate(today, 'YYYY-DDD');
+
+			var relativePrefix = '';
+			if (dayOfYear === dateOfToday) {
+				relativePrefix = t('spreed', 'Today');
+			} else {
+				var yesterday = OC.Util.formatDate(today.subtract(1, 'd'), 'YYYY-DDD');
+
+				if (dayOfYear === yesterday) {
+					relativePrefix = t('spreed', 'Yesterday');
+				} else {
+					relativePrefix = date.fromNow();
+				}
+			}
+
+
+			return relativePrefix + ', ' + OC.Util.formatDate(timestamp, 'LL');
```

Comment:
The whole text should be internationalized, as depending on the locale a different separator (or even order of the prefix and date) could be needed.

Also please add a comment like _'LL' formats a localized date including day of month, month name and year_ to know what it does ;-)